homeassistant/components/template/event.py

```diff
@@ -0,0 +1,235 @@
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import (
+    AddConfigEntryEntitiesCallback,
+    AddEntitiesCallback,
+)
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import TriggerUpdateCoordinator
+from .entity import AbstractTemplateEntity
+from .helpers import (
+    async_setup_template_entry,
+    async_setup_template_platform,
+    async_setup_template_preview,
+)
+from .template_entity import (
+    TEMPLATE_ENTITY_COMMON_CONFIG_ENTRY_SCHEMA,
+    TemplateEntity,
+    make_template_entity_common_modern_attributes_schema,
+)
+from .trigger_entity import TriggerEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Template Event"
+
+CONF_EVENT_TYPE = "event_type"
+CONF_EVENT_TYPES = "event_types"
+
+DEVICE_CLASS_SCHEMA: Final = vol.All(vol.Lower, vol.Coerce(EventDeviceClass))
+
+EVENT_COMMON_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASS_SCHEMA,
+        vol.Required(CONF_EVENT_TYPE): cv.template,
+        vol.Required(CONF_EVENT_TYPES): cv.template,
+    }
+)
+
+EVENT_YAML_SCHEMA = EVENT_COMMON_SCHEMA.extend(
+    make_template_entity_common_modern_attributes_schema(DEFAULT_NAME).schema
+)
+
+
+EVENT_CONFIG_ENTRY_SCHEMA = EVENT_COMMON_SCHEMA.extend(
+    TEMPLATE_ENTITY_COMMON_CONFIG_ENTRY_SCHEMA.schema
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the template event."""
+    await async_setup_template_platform(
+        hass,
+        EVENT_DOMAIN,
+        config,
+        StateEventEntity,
+        TriggerEventEntity,
+        async_add_entities,
+        discovery_info,
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Initialize config entry."""
+    await async_setup_template_entry(
+        hass,
+        config_entry,
+        async_add_entities,
+        StateEventEntity,
+        EVENT_CONFIG_ENTRY_SCHEMA,
+        True,
+    )
+
+
+@callback
+def async_create_preview_event(
+    hass: HomeAssistant, name: str, config: dict[str, Any]
+) -> StateEventEntity:
+    """Create a preview event."""
+    return async_setup_template_preview(
+        hass,
+        name,
+        config,
+        StateEventEntity,
+        EVENT_CONFIG_ENTRY_SCHEMA,
+    )
+
+
+class AbstractTemplateEvent(AbstractTemplateEntity, EventEntity):
+    """Representation of a template event features."""
+
+    _entity_id_format = ENTITY_ID_FORMAT
+
+    # The super init is not called because TemplateEntity and TriggerEntity will call AbstractTemplateEntity.__init__.
+    # This ensures that the __init__ on AbstractTemplateEntity is not called twice.
```

Comment:
If `super()` would be used instead of calling a class method explicitly there will never be a double call since there's a method resolution order.

Generally that's better practice. We should aim for a design that can do that.

Often the easy way to do that is to use named arguments only and also accept arbitrary named arguments that are just passed on to the parent.